### PR TITLE
fix: obey rate limits for all request types correctly

### PR DESCRIPTION
**Pull request by max-wittig**
Waiting for #169 to be merged

### CODE REVIEW

File: test/tests/infrastructure/RequestHelper.js [hunk 1]
```diff
@@ -1,4 +1,7 @@
 import { RequestHelper } from '../../../src/infrastructure';
+import Gitlab from '../../..';
```
Comment (casz) [inline test/tests/infrastructure/RequestHelper.js]: Please fix the import statement: https://travis-ci.org/jdalrymple/node-gitlab/jobs/419127291#L3209-L3210

**Comment by max-wittig** [comment]
I sadly had to drop the tests, as jest runs out of order and in parallel.

File: src/infrastructure/RequestHelper.js [hunk 2]
```diff
@@ -111,46 +113,52 @@ async function getPaginated(service, endpoint, options = {}) {
...
 
-  static post(service, endpoint, options = {}, form = false) {
-    const body = form ? 'formData' : 'body';
+  static async request(type, service, endpoint, options = {}, form = false, stream = false) {
     const requestOptions = defaultRequest(service, endpoint, {
       headers: service.headers,
-      [body]: options,
     });
+    try {
+      if (type === 'get') {
```
Comment (jdalrymple) [inline src/infrastructure/RequestHelper.js]: wouldnt a switch statement here be better?

**Comment by max-wittig** [comment]
@jdalrymple Any blockers? I think the pipeline failure is a random occurrance. Right?

**Comment by jdalrymple** [comment]
Looks like a few of the properties changed in the application settings test. Just need to add the missing props

**Comment by max-wittig** [comment]
@jdalrymple Sorry I haven't found the time to adjust the tests. 
Thanks for the merge :)

**Pull request merged**